corehq/apps/reports/standard/cases/case_data.py

```diff
@@ -555,6 +559,209 @@ def close_case_view(request, domain, case_id):
+                    if any(action in actions for action in update_actions):
+                        if touched_id not in affected_cases:
+                            affected_cases[touched_id] = {}
+                        affected_cases[touched_id][form_id] = ', '.join(actions)
+                        case_names[touched_id] = case_object.name
+            cases[case.case_id][form_id] = case_actions
+        return
+
+    case_instance = safely_get_case(request, domain, case_id)
+    subcase_count = 0
+    try:
+        get_forms_for_deletion_from_case(case_instance, subcase_count)
+    except ValueError:
+        messages.error(request, _("Deleting this case would delete too many related cases. "
+                                  "Please delete some of this cases' subcases before attempting"
+                                  "to delete this case."))
+        return {}, True
+
+    def get_case_link(caseid):
+        url = reverse('case_data', args=[domain, caseid])
+        return mark_safe('<a href="{}"> {} </a>'.format(url, case_names[caseid]))
+
+    def get_form_link(formid):
+        url = reverse('render_form_data', args=[domain, formid])
+        return mark_safe('<a href="{}"> {} </a>'.format(url, form_names[formid]))
+
+    prepared_cases = {}
+    for case in cases:
+        prepared_forms = {}
+        for form in cases[case]:
+            form_link = get_form_link(form)
+            prepared_forms[form_link] = {}
+            for case_action in cases[case][form]:
+                prepared_forms[form_link][case_action] = ', '.join(cases[case][form][case_action])
+        prepared_cases[get_case_link(case)] = prepared_forms
+
+    prepared_reopened_cases = {}
+    for case in reopened_cases:
+        prepared_reopened_cases[get_case_link(case)] = get_form_link(reopened_cases[case])
+
+    prepared_affected_cases = {}
+    for case in affected_cases:
+        if case in delete_cases:
+            continue
+        prepared_forms = {}
+        for form in affected_cases[case]:
+            prepared_forms[get_form_link(form)] = affected_cases[case][form]
+        prepared_affected_cases[get_case_link(case)] = prepared_forms
+
+    return {
+        'main_case_name': case_instance.name,
+        'delete_dict': prepared_cases,
+        'affected_cases': prepared_affected_cases,
+        'reopened_cases': prepared_reopened_cases,
+        'case_delete_list': delete_cases,
+        'form_delete_list': delete_forms,
+    }, False
+
+
+@location_safe
+class DeleteCaseView(BaseProjectReportSectionView):
+    urlname = 'soft_delete_case_view'
+    page_title = gettext_lazy('Delete Case and Related Forms')
+    template_name = 'reports/reportdata/case_delete.html'
+    delete_dict = {}
+
+    @method_decorator(require_case_view_permission)
+    def dispatch(self, request, *args, **kwargs):
+        self.delete_dict, redirect = get_cases_and_forms_for_deletion(request, self.domain, self.case_id)
```

Comment:
How long does this take to run? Wondering it would make sense to offload it to celery

Reply:
From my testing, It doesn’t take long at all. There’s also a limit on how many cases it can look at/return so the lag shouldn’t really be noticeable.